fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/auto/DruidNavigator.java

```diff
@@ -54,14 +53,13 @@ public DruidNavigator(DruidWebService druidWebService) {
     @Override
     public List<? extends DataSourceConfiguration> get() {
         if (tableConfigurations.isEmpty()) {
-            loadAllDatasources();
+            Future<Response> responseFuture = loadAllDatasources();
             LOG.info("Loading all datasources");
             try {
-                //TODO Druid should ideally return a future, but this accomplishes the same result.
-                countDownLatch.await(5, TimeUnit.SECONDS);
-            } catch (InterruptedException e) {
+                responseFuture.get();
```

Comment:
Add a comment in here that indicates we're only calling `get` so that we wait for the response to come back and be processed before continuing execution.